torchtitan/experiments/deepseek_v3/model.py

```diff
@@ -520,6 +513,41 @@ def __init__(self, config):
                 config=config, intermediate_size=intermediate_size
             )
 
+        # Group Gemm
+        self._initialize_group_gemm_strategies()
+
+        assert (
+            self.group_mm in self.group_gemm_strategies
+        ), f"selected group gemm {self.group_mm} is not avaiable!"
+        # keep active gg ready
+        self.group_gemm_instance = self.group_gemm_strategies[self.group_mm]
+        self._buffer_initialized = False
+
+    def _initialize_group_gemm_strategies(self):
+        """Initialize available group GEMM strategies"""
+        self.group_gemm_strategies = {
+            "torch": TorchBF16GroupGEMM(self.activation_function),
+            "torchao": (
+                TorchAOBF16GroupGEMM(self.activation_function)
+                if TorchAOBF16GroupGEMM.is_available()
+                else None
+            ),
+            "torchfp8": (
+                TorchFP8GroupGEMM(self.activation_function)
+                if TorchFP8GroupGEMM.is_available()
+                else None
+            ),
+            "dsgemm": (
+                DSGroupGEMM(self.activation_function)
+                if DSGroupGEMM.is_available()
+                else None
+            ),
+        }
+
+    def zero_print(self, msg):
+        if self.rank == 0:
+            print(f"{msg}")
+
```

Comment:
Maybe use some global function or terminal filtering than putting it in model code?

Reply:
removed, and will implement terminal filtering to be in line with Titan main.